### DISCUSSION TITLE
Most read/popular

### DISCUSSION THREAD

**Discussion by KDederichs** [sulu/sulu#6639] [Q&A] [answered]
Hi,
I was wondering if it's currently possible to somehow track clicks on a page using sulu only to then display popular ones using smart content. I thought about extending the Page entity to do that but that doesn't seem possible atm?

Pointers if this is possible would be appreciated :)

**Comment by alexander-schranz** [comment]
@KDederichs sulu is not tracking anything currently. And as sulu highly uses HttpCache by default Symfony but also in cases Varnish, there isn't even PHP called in that cases. So to track things you mostly will relay on third party things like, google analytics, more popular today is matomo, or a custom tracking script. In all cases you require to track things by using a JS and sulu doesn't force anything in the frontend.

Still the feature is very interesting, it would be nice to sync data like most visits and so on back to the sulu page entity and use that as some kind of sorting in smart content. If I remember correctly Drupal has such kind of feature not sure if it was a plugin or not, there they use some value called radioactivity and depending on that value it was used to show popular articles. I like the the idea so I will convert to a issue.

**Reply by KDederichs** [reply]
Sure, good to know it's not possible atm but if it becomes a feature in the future even better :)

**Comment by alexander-schranz** [comment, marked as answer]
How you could currently achieve it. You track the visits with some kind of javascript or even used popular tools.

And then use a cronjob or hooks to update a pages value in phpcr.

Then you could create an own pages smart content provider: https://github.com/sulu/sulu/blob/8592d999359b5538ec7c97cecae2ba4e045b7de2/src/Sulu/Component/Content/SmartContent/PageDataProvider.php#L169-L177

which will sort by additional field. Following docs could also help: 

https://docs.sulu.io/en/2.4/cookbook/custom-page-data-provider.html

**Reply by KDederichs** [reply]
Yeah I think I'll build a little visitor JS then that'll sync into the phpcr (you don't happen to have docs around on how to interact with that do you? Have not seen any)

**Reply by alexander-schranz** [reply]
Normally you can load and update data of a page using the document manager: https://docs.sulu.io/en/2.4/reference/components/document-manager/fixtures.html

But maybe in this case its better to directly go through PHPCR to update the value in default and liveSession as that would be required that it works also in admin and preview correctly. The uuid by url you get via: https://github.com/sulu/sulu/blob/8592d999359b5538ec7c97cecae2ba4e045b7de2/src/Sulu/Bundle/WebsiteBundle/Routing/ContentRouteProvider.php#L153-L157

You can checkout the jackalope docs. The best docs about changing things through the phpcr session would be in sulu docs this: https://docs.sulu.io/en/2.4/cookbook/migrate-content-data.html.
